plugins/techdocs/api-report.md

```diff
@@ -141,74 +69,101 @@ export type DocsTableRow = {
+    createTypeColumn(): TableColumn<DocsTableRow>;
+  };
+  actions: {
+    createCopyDocsUrlAction(copyToClipboard: Function): (row: DocsTableRow) => {
+      icon: () => JSX.Element;
+      tooltip: string;
+      onClick: () => any;
+    };
+    createStarEntityAction(
+      isStarredEntity: Function,
+      toggleStarredEntity: Function,
+    ): ({ entity }: DocsTableRow) => {
+      cellStyle: {
+        paddingLeft: string;
+      };
+      icon: () => JSX.Element;
+      tooltip: string;
+      onClick: () => any;
+    };
+  };
 };
 
-// Warning: (ae-missing-release-tag) "EntityTechdocsContent" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
-//
-// @public (undocumented)
-export const EntityTechdocsContent: (_props: {}) => JSX.Element;
+// @public
+export type EntityListDocsTableProps = {
+  columns?: TableColumn<DocsTableRow>[];
+  actions?: TableProps<DocsTableRow>['actions'];
+};
 
-// Warning: (ae-missing-release-tag) "isTechDocsAvailable" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
-//
-// @public (undocumented)
+// @public
+export const EntityTechdocsContent: () => JSX.Element;
+
+// @public
 export const isTechDocsAvailable: (entity: Entity) => boolean;
 
-// Warning: (ae-missing-release-tag) "PanelType" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
-//
-// @public (undocumented)
+// @public
+export interface PanelConfig {
+  // (undocumented)
+  description: string;
+  // (undocumented)
+  filterPredicate: ((entity: Entity) => boolean) | string;
+  // (undocumented)
+  panelCSS?: CSSProperties;
+  // (undocumented)
+  panelType: PanelType;
+  // (undocumented)
+  title: string;
+}
+
+// @public
 export type PanelType = 'DocsCardGrid' | 'DocsTable';
 
-// Warning: (ae-forgotten-export) The symbol "Props" needs to be exported by the entry point index.d.ts
-// Warning: (ae-missing-release-tag) "Reader" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
-//
-// @public (undocumented)
-export const Reader: ({
-  entityRef,
-  onReady,
-  withSearch,
-}: Props_3) => JSX.Element;
+// @public
+export const Reader: (props: ReaderProps) => JSX.Element;
+
+// @public
+export type ReaderProps = {
+  entityRef: EntityName;
+  withSearch?: boolean;
+  onReady?: () => void;
+};
 
-// Warning: (ae-missing-release-tag) "Router" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
-//
-// @public (undocumented)
+// @public
 export const Router: () => JSX.Element;
 
 // @public
 export type SyncResult = 'cached' | 'updated';
 
+// @public
+export interface TabConfig {
+  // (undocumented)
+  label: string;
+  // (undocumented)
+  panels: PanelConfig[];
+}
+
+// @public
+export type TabsConfig = TabConfig[];
+
 // @public
 export interface TechDocsApi {
```

Comment:
@emmaindal 

💡 Idea: What do you think of consolidating `TechDocsApi` and `TechDocsStorageApi` by deprecating (and later removing) this one?  On this API in particular:

- `getApiOrigin()` seems like a passthrough to discovery API; we could replace instances of that throughout.
- `getEntityMetadata` seems like a passthrough to catalog API; we could replace instances of that throughout.
- `getTechDocsMetadata` seems like it could be a new method on `TechDocsStorageApi`, since it depends on knowledge about TechDocs' storage location/topology anyway.

Reply:
Hmm, let me think a bit of it. I like the idea of merging the two different ones into one. What I'm concerned about (and let me know if I misunderstood) is to replace usage of `getApiOrigin` and `getEntityMetadata` with the catalog and discovery API. Even if its just pass throughs, it's still clear to me if I implement my own API, what methods I need to provide I guess - for TechDocs to continue to work as expected. If I do not use the entity metadata from the catalog (maybe that is not even a case here but just came to mind), I might use it from somewhere else for example.

Reply:
Yeaaah, you're probably right. There are probably a very small number of folks who don't explicitly rely on the catalog itself, but it's a use-case that's technically supported now that would be difficult or impossible to support if we removed this layer.

Maybe something to consider later down the line, though.

Reply:
Yes agree! This PR is ready now so please take a look again!